clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -240,10 +240,11 @@ private CompletableFuture<Void> maybeAutoCommit(final Map<TopicPartition, Offset
             return CompletableFuture.completedFuture(null);
         }
 
-        CompletableFuture<Void> result = addOffsetCommitRequest(offsets, expirationTimeMs, retryOnStaleEpoch)
-            .whenComplete(autoCommitCallback(offsets));
         autocommit.resetTimer();
         autocommit.setInflightCommitStatus(true);
+
```

Comment:
We have raised a PR to fix this bug: https://github.com/apache/kafka/pull/15282. We can ignore it here.